core/tauri/src/manager.rs

```diff
@@ -471,21 +471,21 @@ impl<R: Runtime> WindowManager<R> {
     }
 
     let window_url = Url::parse(&pending.url).unwrap();
-    let window_origin =
-      if cfg!(windows) && window_url.scheme() != "http" && window_url.scheme() != "https" {
-        format!("https://{}.localhost", window_url.scheme())
-      } else {
-        format!(
-          "{}://{}{}",
-          window_url.scheme(),
-          window_url.host().unwrap(),
-          if let Some(port) = window_url.port() {
-            format!(":{port}")
-          } else {
-            "".into()
-          }
-        )
-      };
+    let window_origin = if window_url.scheme() == "data" {
+      "null".into()
```

Comment:
Though actually wry needs some changes for this. The `href` should be the actual data URL string in this case, instead of http://localhost (and we indeed use null here as origin).

Reply:
I will open a PR at wry for this.

Reply:
The boring thing is we would need to backport it too to land on v1.